presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -384,117 +357,23 @@ private static Expression toExpression(Object value)
         protected Node visitShowPartitions(ShowPartitions showPartitions, Void context)
         {
             QualifiedObjectName table = createQualifiedObjectName(session, showPartitions, showPartitions.getTable());
-            TableHandle tableHandle = metadata.getTableHandle(session, table)
+            metadata.getTableHandle(session, table)
                     .orElseThrow(() -> new SemanticException(MISSING_TABLE, showPartitions, "Table '%s' does not exist", table));
 
-            TableLayout layout = getLayoutFromFilterScanQuery(showPartitions, showPartitions.getTable(), showPartitions.getWhere())
-                    .orElseGet(() -> getLayoutWithoutPartitionsFromMetadata(showPartitions, showPartitions.getTable(), tableHandle));
-            List<Expression> rows = getPartitionsAsValuesRows(showPartitions, table, layout);
-            List<String> partitioningColumns = getPartitioningColumns(showPartitions, table, tableHandle, layout);
-
-            Optional<Expression> where;
-            if (rows.isEmpty()) {
-                // VALUES does not allow no rows
-                List<Expression> row = partitioningColumns.stream()
-                        .map(partitioningColumn -> new NullLiteral())
-                        .collect(toImmutableList());
-                rows = ImmutableList.of(new Row(row));
-                where = Optional.of(FALSE_LITERAL);
-            }
-            else {
-                where = showPartitions.getWhere();
-            }
+            QualifiedObjectName partitionsTable = new QualifiedObjectName(table.getCatalogName(), table.getSchemaName(), table.getObjectName() + "$partitions");
+            metadata.getTableHandle(session, partitionsTable)
+                    .orElseThrow(() -> new SemanticException(NOT_SUPPORTED, showPartitions, "Table '%s' is not partitioned ", table));
 
             return simpleQuery(
-                    selectList(partitioningColumns.stream().map(QueryUtil::identifier).collect(toImmutableList())),
-                    aliased(new Values(rows), "x", partitioningColumns),
-                    where,
+                    selectList(new AllColumns()),
+                    new Table(QualifiedName.of(partitionsTable.getCatalogName(), partitionsTable.getSchemaName(), partitionsTable.getObjectName())),
```

Comment:
There should be some shorter way to convert `QualifiedObjectName` to `QualifiedName` ..

Reply:
`createQualifiedName()` in `MetadataUtil`